stl/inc/ranges

```diff
@@ -872,6 +872,133 @@ namespace ranges {
 
         inline constexpr _Reverse_fn reverse;
     } // namespace views
+
+    // CLASS TEMPLATE ranges::common_view
+    // clang-format off
+    template <view _Vw>
+        requires (!common_range<_Vw> && copyable<iterator_t<_Vw>>)
+    class common_view : public view_interface<common_view<_Vw>> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ _Vw _Base{};
+
+    public:
+        common_view() = default;
+        constexpr explicit common_view(_Vw _Base_) noexcept(is_nothrow_move_constructible_v<_Vw>) // strengthened
+            : _Base(_STD move(_Base_)) {}
+
+        // clang-format off
+        template <viewable_range _Rng>
+            requires (!common_range<_Rng> && constructible_from<_Vw, views::all_t<_Rng>>)
+        constexpr explicit common_view(_Rng&& _Range) noexcept(
+            is_nothrow_constructible_v<_Vw, views::all_t<_Rng>&&>) // strengthened
+            : _Base(views::all(_STD forward<_Rng&&>(_Range))) {}
+        // clang-format on
+
+        _NODISCARD constexpr _Vw base() const& noexcept(
+            is_nothrow_copy_constructible_v<_Vw>) /* strengthened */ requires copy_constructible<_Vw> {
+            return _Base;
+        }
+        _NODISCARD constexpr _Vw base() && noexcept(is_nothrow_move_constructible_v<_Vw>) /* strengthened */ {
+            return _STD move(_Base);
+        }
+
+        _NODISCARD constexpr auto begin() noexcept(
+            _RANGES begin(_Base) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) {
+            if constexpr (random_access_range<_Vw> && sized_range<_Vw>) {
+                return _RANGES begin(_Base);
+            } else {
+                return common_iterator<iterator_t<_Vw>, sentinel_t<_Vw>>{_RANGES begin(_Base)};
+            }
+        }
+
+        _NODISCARD constexpr auto begin() const
+            noexcept(_RANGES begin(_Base) && is_nothrow_move_constructible_v<iterator_t<const _Vw>>) {

```

Comment:
Missing `requires range<const _Vw>` on this overload. (And on 934.)

Reply:
Also missing a layer of `noexcept()` here. (And a few more occurrences.)

Reply:
Done